src/core_plugins/input_control_vis/public/vis_controller.js

```diff
@@ -7,14 +7,22 @@ class VisController {
   constructor(el, vis) {
     this.el = el;
     this.vis = vis;
+    this.lastTimeFilter = JSON.stringify(this.vis.API.timeFilter.time);
     this.controls = [];
 
     this.queryBarUpdateHandler = this.updateControlsFromKbn.bind(this);
     this.vis.API.queryFilter.on('update', this.queryBarUpdateHandler);
   }
 
   async render(visData, status) {
-    if (status.params) {
+    let timeFilterChanged = false;
```

Comment:
Are we going to miss potential new items when using a relative time range, as the `this.vis.API.timeFilter.time` will be equal, but the underlying data ranges will differ?

Reply:
Right now there is no way knowing if the `render` method is getting called because the timefilter updated. I thought comparing the time would work but you are right that when data is fetched for a relative time range then this strategy breaks down. Any ideas on how to handle relative dates?

Reply:
What's the downside to rendering every-time `render` is called? Does this cause the data to be fetched again, and we're trying to prevent this when it's unnecessary?

Reply:
exactly. Each time the component is drawn, then the terms agg or min/max agg will be required.

Reply:
timefilter has a getBounds method call. I can just watch the results of that instead. Also, I am going to move that logic into https://github.com/elastic/kibana/blob/master/src/ui/public/vis/update_status.js. That way the status parameter will just contain a `time` key that will be set to true or false.

Reply:
I think we should rely on Elasticsearch to cache what it can and make our requests as fast as possible. Trying to dedupe things client side (especially per visualization) seems like it will be very tricky to get right in all scenarios if not impossible.

Reply:
> I think we should rely on Elasticsearch to cache what it can and make our requests as fast as possible. Trying to dedupe things client side (especially per visualization) seems like it will be very tricky to get right in all scenarios if not impossible.

Agreed. I'm not aware of us doing this for other visualizations, was there a specific reason why Input Controls should be the exception to this @nreese?

Reply:
Other visualizations do have custom logic to render under certain conditions. Visualizations are passed a [status](https://github.com/elastic/kibana/blob/master/src/core_plugins/input_control_vis/public/vis_controller.js#L16) object to know why render is getting called. For example, the [map visualization](https://github.com/elastic/kibana/blob/master/src/core_plugins/tile_map/public/base_maps_visualization.js#L46) does different things based on the contents of `status`. 

I created a new [PR](https://github.com/elastic/kibana/pull/15856) that adds `time` to the status object so all time change logic can be removed from vis_controller.js